### PR TITLE
Grid & Layout components

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
#6273  

## Description of Changes
This PR introduces two layout components that we will be using for every single page to make sure that the paddings are the same regardless the user's location in the app. 

- PageLayout - component that should be used to every page we build. It's a wrapper that holds the styling for paddings in different breakpoints
- Grid - component that should be used whenever we want to use two column layout 
- [this commit](https://github.com/hicommonwealth/commonwealth/commit/424fafa2067a6cc32e603e8e2b5e5b4c32f75a6a) shows new components implementation: Grid and PageLayout
- [this commit](https://github.com/hicommonwealth/commonwealth/commit/58e9acf040514b61db4907eb20740eca572c0807) shows how to use these components in existing pages (this commit has been reverted as it was only for showcase purpose)

https://github.com/hicommonwealth/commonwealth/assets/14819225/1b762c4e-3624-405e-84bd-1f84b72420de

## Next steps

- **There will be another ticket that handles actual implementation of new components into existing pages.**
- **We might need to rethink how breadcrumbs work (visually) but this also gonna be part of the next ticket.**

### CODE REVIEW

**Comment by zakhap** [comment]
For context, I approve this approach with a full heart. I understand that there will be a lot of elbow grease needed to move all our existing pages to this layout system but it will pay dividends.

File: packages/commonwealth/client/scripts/navigation/GeneralRoutes.tsx [hunk 1]
```diff
@@ -36,6 +37,11 @@ const GeneralRoutes = () => [
     path="/components"
     element={withLayout(ComponentsShowcasePage, { type: 'common' })}
   />,
+  <Route
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/navigation/GeneralRoutes.tsx:40]: i think we can make a new DEV_MODE flag (or check NODE_ENV!='production') or non-user facing pages

**Review by mzparacha — APPROVED** [review]
Looks good. 

Outside scope of this PR: i am not sure if the grid section 2, would look good on all pages in small screens (where it goes below grid section 1 in a row format), prob something to think about 🤔

image

**Review by lzach83 — APPROVED** [review]
Approving, there will be some cleanup that needs to be done with the breadcrumbs. Like @CowMuon mentioned during yesterdays meeting that may happen during the cleanup week.

File: packages/commonwealth/client/scripts/navigation/GeneralRoutes.tsx (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/navigation/GeneralRoutes.tsx]: actually I will get rid of it as it was only for testing purpose

**Comment by masvelio** [comment]
> Looks good.
> 
> Outside scope of this PR: i am not sure if the grid section 2, would look good on all pages in small screens (where it goes below grid section 1 in a row format), prob something to think about 🤔
> 
> image

yes, this is something that we will need to figure out together with the design team on page basis

**Pull request merged**